src/services/solana-client/transferWAudioBalance.ts

```diff
@@ -0,0 +1,172 @@
+import {
+  AccountMeta,
+  Connection,
+  PublicKey,
+  SYSVAR_INSTRUCTIONS_PUBKEY
+} from '@solana/web3.js'
+import BN from 'bn.js'
+import * as borsh from 'borsh'
+import keccak256 from 'keccak256'
+import secp256k1 from 'secp256k1'
+import {
+  solanaClusterEndpoint,
+  tokenProgramKey,
+  audiusProgramPubkey,
+  generatedProgramPDA
+} from './config'
+import SolanaClient from './index'
+
+/**
+ * Transfer wAUDIO between wallets on solana
+ */
+
+class TransferInstructionData {
+  signature: Uint8Array
+  eth_address: Uint8Array
+  recovery_id: number
+
+  constructor({
+    signature,
+    ethAddress,
+    recoveryId
+  }: {
+    signature: Uint8Array
+    ethAddress: Uint8Array
+    recoveryId: number
+  }) {
+    this.signature = signature
+    this.eth_address = ethAddress
+    this.recovery_id = recoveryId
+  }
+}
+
+const transferInstructionSchema = new Map([
+  [
+    TransferInstructionData,
+    {
+      kind: 'struct',
+      fields: [
+        ['signature', [64]],
+        ['eth_address', [20]],
+        ['recovery_id', 'u8']
+      ]
+    }
+  ]
+])
+
+// transferWAudioBalance transfers wrapped Audio from one generated solana account to another.
+// For it to work, you have to have the eth private key belonging to the eth public key
+// that generated the solana account
+export async function transferWAudioBalance(
+  senderEthAddress: string,
```

Comment:
we should probably type these more specifically - something akin to how we dealt with the token types in the wallet client w the Brand type? wdyt